src/io/rainrobot/maxdispachreq/AvailabilityService.java

```diff
@@ -0,0 +1,61 @@
+package io.rainrobot.maxdispachreq;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class AvailabilityService {
+
+    private int maxAvailability;
+    private Map<Integer, Set<ServerAddress>> avelabiletyServerSetMap;
+    private AddressAvelabiletyTracker addressAvelabiletyTracker;
+
+    private void recomputeMaxAvailability() {
+        if (getMaxAvailabilitySet().size() == 0) {
+            do{
+                maxAvailability --;
+            } while (getMaxAvailabilitySet().size() == 0
+                                    && maxAvailability != 1);
+            maxAvailability --;
+        }
+    }
+    
+    private Set<ServerAddress> getMaxAvailabilitySet() {
+        return avelabiletyServerSetMap.get(maxAvailability);
+    }
+
+    public ServerAddress popFromMaxAvailableSet() {
+        ServerAddress popServer = getMaxAvailabilitySet().iterator().next();
+        decrementAddressAvailability(popServer, maxAvailability);
+        recomputeMaxAvailability();
+        return popServer;
+    }
+
+    private void decrementAddressAvailability(ServerAddress popServer, int availability) {
```

Comment:
logic is not clear, what are you doing here. how the availability service works, what is max requests and what is availability?